test/Main.hs

```diff
@@ -29,6 +57,20 @@ main = do
       it "should not have the same output as the input" $
         property testNotDifference
     describe "Binary AND function" $ do
-      it "returns the input value if both are the same" $
+      it "returns 'I' if both inputs are 'I'" $
```

Comment:
This looks like a regression to me. The new description is more narrow, and thus says less about the properties of the function.

Reply:
Is the description not more characteristic this way? AND returns 'I' only if both inputs are 'I', otherwise the inputs don't matter.